tests/amqpprox_session.t.cpp

```diff
@@ -365,6 +366,26 @@ void SessionTest::testSetupProxyOutOfOrderOpen(int idx)
     });
 }
 
+void SessionTest::testSetupProxyReceiveClose(int idx)
+{
+    // Client  <--------Close-----  Proxy  <--------Close------  Broker
+    methods::Close receivedClose;
+    receivedClose.setReply(123, "Broker is closing the connection");
+    d_clientState.pushItem(idx, Data(encode(receivedClose)));
+
+    d_serverState.expect(idx, [this, receivedClose](const auto &items) {
+        auto data = filterVariant<Data>(items);
+        ASSERT_EQ(data.size(), 1);
+        EXPECT_EQ(data[0], Data(encode(receivedClose)));
+        EXPECT_THAT(items, Contains(VariantWith<Call>(Call("shutdown"))));
+        EXPECT_THAT(items, Contains(VariantWith<Call>(Call("close"))));
+    });
+    d_clientState.expect(idx, [this](const auto &items) {
```

Comment:
and the expectation is here for the receivedClose?

Reply:
As I mentioned earlier, this state represents interaction with server. So we will not have that expectation.